app/models/spree/gateway/stripe_gateway.rb

```diff
@@ -9,6 +9,9 @@ class Gateway::StripeGateway < Gateway
       'Visa' => 'visa'
     }
 
+    class_attribute :currencies_without_fractions
+    self.currencies_without_fractions = %w(BIF BYR CLP CVE DJF GNF HUF ISK JPY KMF KRW PYG RWF UGX VND VUV XAF XOF XPF)
```

Comment:
how about using `provider_class.currencies_without_fractions` in `def non_fractional_currency?`?